### PR TITLE
ar71xx: add support for OCEDO Raccoon

### PR DESCRIPTION
**Pull request by blocktrron**
This commit adds support for the OCEDO Raccoon

SOC:	Atheros AR9344
RAM:    128MB
FLASH:  16MiB
WLAN1:  AR9344 2.4 GHz 802.11bgn 2x2
WLAN2:  AR9382 5 GHz 802.11an 2x2
INPUT:  RESET button
LED:    Power, LAN, WiFi 2.4, WiFi 5
Serial: Header Next to Black metal shield
        Pinout is 3.3V - GND - TX - RX (Arrow Pad is 3.3V)
        The Serial setting is 115200-8-N-1.

	NOTE: The U-Boot won't boot with the serial attached.
	Boot the device without serial attached and attach it
	after 3 seconds.

Tested and working:
 - Ethernet
 - 2.4 GHz WiFi
 - 5 GHz WiFi
 - TFTP boot from ramdisk image
 - Installation via ramdisk image
 - OpenWRT sysupgrade
 - Buttons
 - LEDs

Installation seems to be possible only through booting an OpenWRT
ramdisk image.

Hold down the reset button while powering on the device. It will load a
ramdisk image named 'raccoon-uImage-initramfs-lzma.bin' from 192.168.100.8.

Wait for the ramdisk image to boot. OpenWRT can be written to the flash
via sysupgrade or mtd.

Due to the flip-flop bootloader which we not (yet) support, you need to
set the partition the bootloader is selecting. It is possible from the
initramfs image with

 > fw_setenv bootcmd run bootcmd_1

Afterwards you can reboot the device.

Signed-off-by: David Bauer <mail@david-bauer.net>

### CODE REVIEW

**Comment by herrbett** [comment]
We testet the PR on a few of our Devices.
It works flawlessly.
Thank you for your Work. 👍

File: target/linux/ar71xx/files/arch/mips/ath79/machtypes.h [hunk 1]
```diff
@@ -182,6 +182,7 @@ enum ath79_mach_type {
 	ATH79_MACH_PB44,			/* Atheros PB44 reference board */
 	ATH79_MACH_PQI_AIR_PEN,			/* PQI Air Pen */
 	ATH79_MACH_QIHOO_C301,			/* Qihoo 360 C301 */
+	ATH79_MACH_RACCOON,			/* Ocedo Raccoon */
```
Comment (blogic) [inline target/linux/ar71xx/files/arch/mips/ath79/machtypes.h]: alphabetical ordering

File: target/linux/ar71xx/generic/config-default [hunk 2]
```diff
@@ -130,6 +130,7 @@ CONFIG_ATH79_MACH_PB42=y
 CONFIG_ATH79_MACH_PB44=y
 CONFIG_ATH79_MACH_PQI_AIR_PEN=y
 CONFIG_ATH79_MACH_QIHOO_C301=y
+CONFIG_ATH79_MACH_RACCOON=y
```
Comment (blogic) [inline target/linux/ar71xx/generic/config-default]: ...

File: target/linux/ar71xx/image/generic-senao.mk [hunk 3]
```diff
@@ -39,3 +39,15 @@ define Device/koala
   IMAGE/sysupgrade.bin := append-kernel | pad-to $$$$(KERNEL_SIZE) | append-rootfs | pad-rootfs | append-metadata | check-size $$$$(IMAGE_SIZE)
 endef
 TARGET_DEVICES += koala
+
+define Device/raccoon
```
Comment (mkresin) [inline target/linux/ar71xx/image/generic-senao.mk]: Please have a look at my remarks to https://github.com/openwrt/openwrt/pull/1016. The very same applies here.

**Review by mkresin — CHANGES_REQUESTED** [review]

**Comment by blocktrron** [comment]
@mkresin adjusted to match your requests in #1016

**Review by mkresin — APPROVED** [review]

**Comment by mkresin** [comment]
I merged ath79 patch for the same board instead. I see more no than a little chance that this PR will be ever merged. Hence I'll close the PR.

If you disagree with the my decision, please start a discussion on the OpenWrt development mailinglist.

**Pull request closed without merging**